src/visualizers/VisualizerCycles.ts

```diff
@@ -0,0 +1,270 @@
+import {VisualizerExportModule} from '@/visualizers/VisualizerInterface'
+import {VisualizerDefault} from './VisualizerDefault'
+import {floorSqrt} from '@/shared/math'
+
+/** md
+# cycles
+This visualizer displays the number of steps each sequence value 
+takes to reach a repeating
```

Comment:
Bug report: on A072601, with 1000 terms, it works great.  With 5000 terms, the mouseover bar moves down to the bottom of the screen but there aren't actually any more terms shown.  This sequence does have that many terms.

Reply:
Well, what happens for me now is that if I do 5000 terms on A072601 leaving the cells per row at 64, then there are about 3700 cells visible, and then the blue value bar at the bottom of the canvas. And then if you mouseover the blue bar or even below the canvas altogether, you can see the values past 3700 in the blue bar, even though there are no displayed cells in this range -- they are either under the blue bar or off the bottom of the canvas. And those "virtual" cells go exactly to 5000 and they all seem to "work" fine other than being invisible. Kate, if you can replicate that behavior and you think it's OK at least for this pass, then feel free to resolve this conversation. If you think that the Cycles visualizer should do something else, like recognize that the cell array is going to go past the available space and cap the number of cells displayed to what's actually visible, let Clyde know.